### ISSUE TITLE
The fixed positioning strategy calculates an incorrect position if the containing block isn't the viewport

### ISSUE THREAD

**Issue by Cipscis** [floating-ui/floating-ui#2386]
**Describe the bug**
There are some contexts in which an element's **containing block**, which is used to calculate its position when using `position: fixed;` is not the viewport. One example of this is when an element has an ancestor with `container-type` set to a value like `inline-size`, which is necessary in order to use `@container` queries.

However, it looks like when PopperJS is using the "fixed" positioning strategy it always calculates a popper's position vector based on the assumption that the containing block is the viewport. In contexts where that assumption is not correct, this often results in poppers that are not positioned correctly relative to their reference elements.

**To Reproduce**
1. Go to [this reproduction on Codepen](https://codepen.io/Cipscis/pen/LYXNwYJ)
2. Click on the "Toggle container" button to toggle a `container-type` rule on the main body element
4. See how the popper moves to the incorrect position, based on the position of the containing block element, when it is set to use `container-type: inline-size;`

**Expected behavior**
Poppers using the "fixed" positioning strategy should appear in the correct position relative to their reference elements regardless of what their containing block is.

**Context:**
 - OS: Windows
 - Browser: Chrome
 - Version: 114.0.5735.134

**Comment by atomiks** [comment]
We have now released [Floating UI v1](https://floating-ui.com/docs/motivation), the successor to Popper.

Don't worry, Popper 2 will continue to receive new releases with bug/security fixes provided by external contributors, but development of new features is not continuing at this point. We are also closing bugs related to it as us maintainers are focusing our energy on tracking issues with Floating UI, however you are welcome to create new pull requests to fix bugs in Popper, which we will review and merge, as many packages (such as MUI, Bootstrap, etc) still rely on it. 

**Issues with `fixed` positioning in Popper v2 have been fixed in Floating UI, but you are welcome to create a new PR to fix it in Popper as well.**

Thanks!
